core/selection.js

```diff
@@ -1981,7 +1981,7 @@
 			}
 
 			// Handle special case - fake selection of table cells.
-			if ( editor && editor.config.tableImprovements && isTableSelection( ranges ) && !isSelectingTable ) {
+			if ( editor && editor.plugins.tableselection && isTableSelection( ranges ) && !isSelectingTable ) {
```

Comment:
This conditional is leaking. Using element path in IE 8 causes the call of `performFakeTableSelection`. It's caused by the fact that `editor.plugins.tableselection` is always true (it's an object set **before** plugin's init and not removed).